services/shhext/api.go

```diff
@@ -469,64 +467,22 @@ func (api *PublicAPI) ConfirmMessagesProcessedByID(messageIDs [][]byte) error {
 
 // SendPublicMessage sends a public chat message to the underlying transport
 func (api *PublicAPI) SendPublicMessage(ctx context.Context, msg chat.SendPublicMessageRPC) (hexutil.Bytes, error) {
-	privateKey, err := api.service.w.GetPrivateKey(msg.Sig)
-	if err != nil {
-		return nil, err
-	}
-
-	// This is transport layer agnostic
-	protocolMessage, err := api.service.protocol.BuildPublicMessage(privateKey, msg.Payload)
-	if err != nil {
-		return nil, err
-	}
-
-	symKeyID, err := api.service.w.AddSymKeyFromPassword(msg.Chat)
+	message, err := api.service.InitializePublicMessage(msg.Sig, msg.Chat, msg.Payload, false)
 	if err != nil {
 		return nil, err
 	}
 
-	// Enrich with transport layer info
-	whisperMessage := chat.PublicMessageToWhisper(msg, protocolMessage)
-	whisperMessage.SymKeyID = symKeyID
-
-	// And dispatch
-	return api.Post(ctx, whisperMessage)
+	return api.Post(ctx, *message)
 }
 
 // SendDirectMessage sends a 1:1 chat message to the underlying transport
 func (api *PublicAPI) SendDirectMessage(ctx context.Context, msg chat.SendDirectMessageRPC) (hexutil.Bytes, error) {
-	if !api.service.pfsEnabled {
-		return nil, ErrPFSNotEnabled
-	}
-	// To be completely agnostic from whisper we should not be using whisper to store the key
-	privateKey, err := api.service.w.GetPrivateKey(msg.Sig)
+	message, err := api.service.InitializeDirectMessage(msg.Sig, msg.PubKey, msg.DH, msg.Payload)
```

Comment:
```suggestion
	message, err := api.service.CreateDirectMessage(msg.Sig, msg.PubKey, msg.DH, msg.Payload)
```

Maybe it's just me but initialization suggests that the message is already created but that's not a case here.